docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -56,86 +56,310 @@ You build recursive zkApps with [ZkProgram](/zkapps/o1js-reference/modules/Exper
+  next
+  svelte
+  nuxt
+  empty
+❯ none
+```
+
+The expected output is:
+
+```sh
+✔ Create an accompanying UI project too? · none
+✔ UI: Set up project
+✔ Initialize Git repo
+✔ Set up project
+✔ NPM install
+✔ NPM build contract
+✔ Set project name
+✔ Git init commit
+
+Success!
 
-      method(state: Field) {
-        state.assertEquals(Field(0));
-      },
-    },
+Next steps:
+  cd 09-recursion
+  git remote add origin <your-repo-url>
+  git push -u origin main
+```
+
+The `zk project` command creates the `09-recursion` directory that contains the scaffolding for your project, including tools such as the Prettier code formatting tool, the ESLint static code analysis tool, and the Jest JavaScript testing framework.
+
+1. Change into the `09-recursion` directory and list the contents:
+
+```sh
+$ cd 09-recursion
+$ ls
+```
+
+The output shows these results:
+
+```sh
+LICENSE
+README.md
+babel.config.cjs
+build
+config.json
+jest-resolver.cjs
+jest.config.js
+keys
+node_modules
+package-lock.json
+package.json
+src
+tsconfig.json
+```
+
+For this tutorial, you run commands from the root of the `09-recursion` directory as you work in the `src` directory on files that contain the TypeScript code for the smart contract. Each time you make updates, then build or deploy, the TypeScript code is compiled into JavaScript in the `build` directory.
+
+### Prepare the project
+
+Start by deleting the default files that come with the new project.
+
+1. To delete the old files:
+
+```sh
+$ rm src/Add.ts
+$ rm src/Add.test.ts
+$ rm src/interact.ts
+```
+
+1. Now, create the new files for your project:
+
+```sh
+$ zk file src/Add
+$ touch src/main.ts
+```
+
+- The `zk file` command created the `src/Add.ts` and `src/Add.test.ts` test files.
+- However, this tutorial does not include writing tests, so you just use the `main.ts` file as a script to interact with the smart contract and observe how it works.
+
+In later tutorials, you learn how to interact with a smart contract from the browser, like a typical end user.
+
+3. Now, open `src/index.ts` in a text editor and change it to look like:
+
+```ts src/index.ts
+1 import { Add } from './Add.js';
+2
+3 export { Add };
+```
+
+The `src/index.ts` file contains all of the exports you want to make available for consumption from outside your smart contract project, such as from a UI.
+
+## Write the ZkProgram
+
+Now, the fun part! Write your smart contract in the `src/Add.ts` file.
+
+Line numbers are provided for convenience. A final version of the smart contract is provided in the [Add.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/Add.ts) example file.
+
+This part of the tutorial walks you through the Add ZkProgram code already completed in the `src/Add.ts` example file.
+
+### Copy the example
+
+This tutorial describes each part of the completed code in the [Add.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/Add.ts) example file.
```

Comment:
```suggestion
### Copy the example

Use the Add code already completed in the [Add.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/Add.ts) example file.
```

Reply:
if we need the Add.ts file, put an updated file into the https://github.com/o1-labs/docs2/tree/main/examples/zkapps/09-recursion/src directory